crates/sail-plan/src/function/aggregate.rs

```diff
@@ -1,116 +1,195 @@
 use std::collections::HashMap;
-use std::sync::Arc;
 
 use datafusion::functions_aggregate::{
-    array_agg, average, bit_and_or_xor, bool_and_or, correlation, count, covariance, first_last,
-    grouping, median, min_max, regr, stddev, sum, variance,
+    approx_distinct, approx_percentile_cont, array_agg, average, bit_and_or_xor, bool_and_or,
+    correlation, count, covariance, first_last, grouping, kurtosis_pop, median, min_max, regr,
+    stddev, sum, variance,
 };
+use datafusion_common::ScalarValue;
+use datafusion_expr::expr;
 use datafusion_expr::expr::AggregateFunction;
-use datafusion_expr::{expr, AggregateUDF};
+use datafusion_expr::sqlparser::ast::NullTreatment;
 use lazy_static::lazy_static;
 
 use crate::error::{PlanError, PlanResult};
+use crate::function::common::AggFunction;
+use crate::utils::ItemTaker;
 
 lazy_static! {
-    static ref BUILT_IN_AGGREGATE_FUNCTIONS: HashMap<&'static str, Arc<AggregateUDF>> =
+    static ref BUILT_IN_AGGREGATE_FUNCTIONS: HashMap<&'static str, AggFunction> =
         HashMap::from_iter(list_built_in_aggregate_functions());
 }
 
-struct AggregateFunctionBuilder;
```

Comment:
Is there any reason for removing this? I feel it's more verbose to have `_agg` in `F::default_agg`, `F::custom_agg`, etc. Also the function builder may have too much responsibility when we introduce window functions and table functions.